### ISSUE TITLE
NativeWind v4  - Dark theme toggle with ui.shadcn.com generated CSS variables works only once

### ISSUE THREAD

**Issue by mrzachnugent** [nativewind/nativewind#702]
**Describe the bug**
Toggling the theme with CSS variables generated from ui.shadcn.com only works once. Then, additional toggles do not update the styling based on the theme.  I did get it to work by changing, in the global css file, `.dark {` to `.dark:root` for the variable. I found this from looking at the test cases.

**To Reproduce**
1. In the global CSS file, add variables in `@layer base{}` for `:root{}` and that same variables with different values in `.dark{}`
**Example**: 
```js
@layer base {
  :root {
    --background: 0 0% 100%;
  }
  .dark {
    --background: 240 10% 4%
  }
}
```
2. In the `tailwind.config.js`, add the colors that are extended by the theme
```js
colors: {
  background: 'hsl(var(--background))'
}
```
3. Add a button that uses the `toggleColorScheme` from the `useColorScheme` hook.
4. Add a `<Text className="text-background">Text from theme</Text>`
5. Press on that button multiple times. If you refresh the Expo app, it will change the color of the text on the first press and each additional toggle will not change the color of the text.

**Expected behavior**
It is expected to change the color of the `Text` component on each press of the button that toggles the theme.

**Additional context**
Given the simple fix of changing `.dark {` to `.dark:root`, it might be a question of updating the docs. Please let me know if I should add a minimal reproduction repo to help. Thanks.

**Comment by 592da** [comment]
Having same issue,

also tried to use vars like the docs, and not working

```ts
export const defaultTheme = {
  light: vars({
    "--background": "0 0% 100%",
    "--foreground": "222.2 47.4% 11.2%",
    "--muted": "210 40% 96.1%",
    "--muted-foreground": "215.4 16.3% 46.9%",
    "--popover": "0 0% 100%",
    "--popover-foreground": "222.2 47.4% 11.2%",
    "--border": "214.3 31.8% 91.4%",
    "--input": "214.3 31.8% 91.4%",
    "--card": "0 0% 100%",
    "--card-foreground": "222.2 47.4% 11.2%",
    "--primary": "222.2 47.4% 11.2%",
    "--primary-foreground": "210 40% 98%",
    "--secondary": "210 40% 96.1%",
    "--secondary-foreground": "222.2 47.4% 11.2%",
    "--accent": "210 40% 96.1%",
    "--accent-foreground": "222.2 47.4% 11.2%",
    "--destructive": "0 100% 50%",
    "--destructive-foreground": "210 40% 98%",
    "--ring": "215 20.2% 65.1%",
    "--radius-sm": "0.5rem",
    "--radius": "1rem",
    "--radius-md": "3rem",
    "--radius-lg": "2rem",
  }),
  dark: vars({
    "--background": "224 71% 4%",
    "--foreground": "213 31% 91%",
    "--muted": "223 47% 11%",
    "--muted-foreground": "215.4 16.3% 56.9%",
    "--accent": "216 34% 17%",
    "--accent-foreground": "210 40% 98%",
    "--popover": "224 71% 4%",
    "--popover-foreground": "215 20.2% 65.1%",
    "--border": "216 34% 17%",
    "--input": "216 34% 17%",
    "--card": "224 71% 4%",
    "--card-foreground": "213 31% 91%",
    "--primary": "210 40% 98%",
    "--primary-foreground": "222.2 47.4% 1.2%",
    "--secondary": "222.2 47.4% 11.2%",
    "--secondary-foreground": "210 40% 98%",
    "--destructive": "0 63% 31%",
    "--destructive-foreground": "210 40% 98%",
    "--ring": "216 34% 17%",
    "--radius-sm": "0.5rem",
    "--radius": "1rem",
    "--radius-md": "3rem",
    "--radius-lg": "2rem",
  }),
};

```


```ts title="ThemeProvider"
const ApplicationThemeProvider = ({ children }) => {
  const { setColorScheme, colorScheme } = useColorScheme();
  const mode = useAppSelector((s) => s.system.theme);

  useEffect(() => {
    StatusBar.setBarStyle(mode === "dark" ? "light-content" : "dark-content");
    setColorScheme(mode);
  }, [mode]);

  const template = useMemo(() => getTheme(mode), [mode]);

  return (
    <PaperProvider theme={template}>
      <View style={[defaultTheme[colorScheme], StyleSheet.absoluteFill]}>
        <Headers />
        {children}
      </View>
    </PaperProvider>
  );
};
````

and added default values in tailwindccss config in plugins
```
    ({ addBase }) =>
      addBase({
        ":root": {
          "--background": "0 0% 100%",
          "--foreground": "222.2 47.4% 11.2%",
          "--muted": "210 40% 96.1%",
          "--muted-foreground": "215.4 16.3% 46.9%",
          "--popover": "0 0% 100%",
          "--popover-foreground": "222.2 47.4% 11.2%",
          "--border": "214.3 31.8% 91.4%",
          "--input": "214.3 31.8% 91.4%",
          "--card": "0 0% 100%",
          "--card-foreground": "222.2 47.4% 11.2%",
          "--primary": "222.2 47.4% 11.2%",
          "--primary-foreground": "210 40% 98%",
          "--secondary": "210 40% 96.1%",
          "--secondary-foreground": "222.2 47.4% 11.2%",
          "--accent": "210 40% 96.1%",
          "--accent-foreground": "222.2 47.4% 11.2%",
          "--destructive": "0 100% 50%",
          "--destructive-foreground": "210 40% 98%",
          "--ring": "215 20.2% 65.1%",
          "--radius-sm": "0.5rem",
          "--radius": "1rem",
          "--radius-md": "3rem",
          "--radius-lg": "2rem",
        },
      })
      ```

**Comment by FelipeJz** [comment]
How are you using shadcnui if it wasn't made for react native

**Comment by mrzachnugent** [comment]
> How are you using shadcnui if it wasn't made for react native

I meant the css variables that are from shadcn. If you were to copy paste it to work with nativewind.

And I started re-creating something very similar to ui/shadcn for react native (still a work in progress)

 https://github.com/mrzachnugent/react-native-reusables

**Comment by FelipeJz** [comment]
Amazing, you are just amazing. That is exactly what i've been looking for, thank you!

**Comment by yairhaimo** [comment]
how would you use the custom css variable colors inside your code (navigation headerStyle color for example)?

**Comment by is-it-ayush** [comment]
> how would you use the custom css variable colors inside your code (navigation headerStyle color for example)?

It's just a guess as I haven't tested it but it might work.
- Add your custom css color as a variable to your `globals.css` or equivalent style file. This style file should be imported in your code at layout. For example,
```css
:root {
  --my-custom-color-var: '#ff0000"
}
```
- Then in your headerStyle,
```tsx
headerStyle: {
  backgroundColor: 'var(--my-custom-color-var)',
}
```

**Comment by CuriosBasant** [comment]
Any updates on this?

**Comment by krausediego** [comment]
> Any updates on this? In my case, only RGB values are working and not HSL.

Check your version of nativewind, for me in 4.0.1 the hsl didn't work, but in 4.0.33 it's working.

```json
"nativewind": "^4.0.33"
```

**Comment by danstepanov** [comment]
We are closing all pre-v4.1 issues now that v4.1 is released. If the problem is still happening with the latest v4.1, [please open a new issue](https://github.com/nativewind/nativewind/blob/next/contributing.md#opening-an-issue) and reference this one in the description.
